alerts/cluster-monitoring-operator/TelemeterClientFailures.md

```diff
@@ -0,0 +1,75 @@
+# TelemeterClientFailures
+
+## Meaning
+
+The alert `TelemeterClientFailures` is triggered when the Telemeter client fails
+to send Telemetry data at a certain rate over a period of time
+to Red Hat.
+
+The `telemeter-client-xxxxxxxxx-xxxxx` pod running in the `openshift-monitoring`
+namespace collects [selected platform metrics](https://docs.openshift.com/container-platform/latest/support/remote_health_monitoring/showing-data-collected-by-remote-health-monitoring.html#showing-data-collected-from-the-cluster_showing-data-collected-by-remote-health-monitoring)
+from the `prometheus-k8s` service at
+regular intervals using the `/federate` endpoint and ships them periodically
+to Red Hat using a custom secured protocol.
+
+## Impact
+
+When the alert fires, Red Hat support and engineering teams don't have a complete
+view of the cluster anymore. It may hinder the ability for Red Hat to
+proactively detect issues in the cluster.
+
+## Diagnosis
+
+Review the logs for the pod `telemeter-client-xxxxxxxxx-xxxxx`
+in the `openshift-monitoring` namespace
+
+You can review the logs for the `telemeter-client-xxxxxxxxx-xxxxx` pod in the
+`openshift-monitoring` namespace by running the following command:
+
+```console
+$ oc logs -n openshift-monitoring deployment.apps/telemeter-client -c telemeter-client | less
+```
+
+## Mitigation
+
+* From `4.16` there is metric `metricsclient_http_requests_total` which can be
+  used to identify which side the issue is happening. For older cluster versions
+ `metricsclient_request_retrieve` and `metricsclient_request_send` metrices
+  can be used to check the server status.
```

Comment:
I would give PromQL expressions to guide the users (not tested):
* 4.17+
  * `sum by(client, status_code) (rate(metricsclient_http_requests_total{status_code!~"2.."})) > 0` 
  * Explain that the "client" label value shows which side is faulty
    * `federate_from` when the error is from Prometheus
    * `federate_to` when the error is from Telemeter server
* 4.16 and below
  * `sum by(client, status_code) (rate(metricsclient_request_retrieve{status_code!~"2.."})) > 0` (Prometheus)
  * `sum by(client, status_code) (rate(metricsclient_request_send{status_code!~"2.."})) > 0` (Telemeter server)

Also we're still in `Diagnosis` here.

Reply:
Checked the query in console its giving error, working on it

Reply:
Issue was 

```
1:35: parse error: expected type range vector in call to function "rate", got instant vector
```

updated query to range vector also `!~"2.."` was not working I tried with `=` as well just to make sure some results return